File: server/ent/schema/badge.go
package schema

import (
	"edumeet/ent/schema/trait"
	"edumeet/utils"

	"entgo.io/ent"
	"entgo.io/ent/schema/edge"
	"entgo.io/ent/schema/field"
)

type Badge struct {
	ent.Schema
}

func (Badge) Mixin() []ent.Mixin {
	return []ent.Mixin{
		trait.Blamable{},
	}
}

func (Badge) Fields() []ent.Field {
	ulid := utils.ULID{}
	return []ent.Field{
		field.String("id").DefaultFunc(ulid.GenerateUlid()).Unique(),
		field.String("name"),
		field.String("svg"),
		field.Int("nbRequirementEvent"),
		field.String("type"),
	}
}

func (Badge) Edges() []ent.Edge {
	return []ent.Edge{
		edge.From("users", User.Type).
			Ref("badges"),
	}
}

File: client/lib/core/models/badge.dart
class Badge {
  final String id;
  final String name;
  final String type;
  final int nbRequirementEvent;
  final String svg;

  Badge({
    required this.id,
    required this.name,
    required this.type,
    required this.nbRequirementEvent,
    required this.svg,
  });

  factory Badge.fromJson(Map<String, dynamic> json) {
    return Badge(
      id: json['id'] as String,
      name: json['name'] as String,
      type: json['type'] as String,
      nbRequirementEvent: json['nbRequirementEvent'] as int,
      svg: json['svg'] as String,
    );
  }
}
